src/test_l3_dsDisplay.c

```diff
@@ -0,0 +1,330 @@
+
+    if (status != dsERR_NONE) {
+        UT_LOG_ERROR("dsGetEDID failed with error: %d", status);
+        free(edid);
+        return;
+    }
+
+    UT_LOG_INFO("Result dsGetEDID(dsDisplayEDID_t(productCode:[%d], serialNumber:[%d], manufactureYear:[%d], manufactureWeek:[%d], hdmiDeviceType:[%s], isRepeater:[%s], physicalAddressA:[%u], physicalAddressB:[%u], physicalAddressC:[%u], physicalAddressD:[%u], numOfSupportedResolution:[%d], monitorName:[%s])", \
+             edid->productCode, edid->serialNumber, edid->manufactureYear, edid->manufactureWeek, edid->hdmiDeviceType, edid->isRepeater, edid->physicalAddressA, edid->physicalAddressB, edid->physicalAddressC, edid->physicalAddressD, edid->numOfSupportedResolution, edid->monitorName);
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+   
+    free(edid);
+}
+
+void dsDisplay_GetEDIDBytes()
+{
+    dsError_t status = dsERR_NONE;
+    intptr_t displayHandle =-1;
+    int length = 0;
+    unsigned char *edid = NULL;
+  
+    dsGetDisplay_getHandle();
+  
+    UT_LOG_INFO("Calling dsGetEDID(Handle:[0x%0X]])", displayHandle);
+    status = dsGetEDIDBytes(displayHandle, edid, &length);
+    
+    UT_LOG_INFO("Result dsGetEDIDBytes()  dsError_t=[%s]", mapKeyToString(errorMappingTable, status));
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+
+}
+
+void GetDisplay_AspectRatio()
+{
+    dsError_t status   = dsERR_NONE;
+    dsVideoAspectRatio_t displayAspectRatio;
+    intptr_t displayHandle =-1;
+  
+    dsGetDisplay_getHandle();
+
+    UT_LOG_INFO("Calling dsGetEDID(Handle:[0x%0X]) ", displayHandle);
+    status = dsGetDisplayAspectRatio(displayHandle, &displayAspectRatio);
+
+    UT_LOG_INFO("Result dsGetDisplayAspectRatio(dsVideoAspectRatio_t:[%s], dsError_t=[%s])", mapKeyToString(dsVideoAspect_RatioMappingTable, displayAspectRatio), mapKeyToString(errorMappingTable, status));
+
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+ 
+}
+
+void RegisterDisplayEventCallback()
+{
+    dsError_t status   = dsERR_NONE;
+    intptr_t displayHandle =-1;
+
+    dsGetDisplay_getHandle();
+
+    UT_LOG_INFO("Calling dsGetEDID(Handle:[0x%0X]) ", displayHandle);
+
+    status = dsRegisterDisplayEventCallback(displayHandle,(dsDisplayEventCallback_t)DisplayEventCallback);
+
+    UT_LOG_INFO("Result dsRegisterDisplayEventCallback() dsError_t=[%s]",  mapKeyToString(errorMappingTable, status));
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+  
+}
+
+static UT_test_suite_t * pSuite = NULL;
+
+/**
+ * @brief Register the main tests for this module
+ *
+ * @return int - 0 on success, otherwise failure
+ */
+
+int test_l3_dsDisplay_register (void)
+{
+    /* add a suite to the registry */
+    pSuite = UT_add_suite( "[L3 dsDisplay]", NULL, NULL );
+    if ( NULL == pSuite)
+    {
+        return -1;
+    }
+
+    //Create the test suite
+    UT_add_test( pSuite, "dsDisplay Init", dsDisplay_Init);
+    UT_add_test( pSuite, "dsDisplay Term", dsDisplay_Term);
+    UT_add_test( pSuite, "dsGetDisplay getHandle", dsGetDisplay_getHandle);
```

Comment:
This is not a external functionality, Please remove this as a test.

Reply:
Done